photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 """
 
 from __future__ import division
+import warnings
 import numpy as np
+import copy
 from astropy.table import Table
 from astropy.modeling import models, Parameter, Fittable2DModel
 from astropy.modeling.fitting import LevMarLSQFitter
 from astropy.nddata.utils import subpixel_indices
 from ..utils import mask_to_mirrored_num
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
+    """
+    Used to undicate that a :py:class:`FittableImageModel2D` model is
+    non-normalizable.
+
+    """
+    pass
+
+
+class FittableImageModel2D(Fittable2DModel):
```

Comment:
I think this name should just be `FittableImageModel` instead of `FittableImageModel2D`.  To me, "image" automatically implies "2D", so it seems repetitive.  Some of the other models have `2D` in them (like `Gaussian2D`) but only because there's also a corresponding `1D` version.  So I think the `2D` is only needed where there's ambiguity, and here there's no such thing as a 1D image.